modules/inflowwind/src/InflowWind_Subs.f90

```diff
@@ -269,6 +269,105 @@ SUBROUTINE InflowWind_ParseInputFileInfo( InputFileData, InFileInfo, PriPath, In
 
    CALL ParseAry( InFileInfo, CurLine, 'WindVziList', InputFileData%WindVziList, InputFileData%NWindVel, TmpErrStat, TmpErrMsg, UnEc )
    if (Failed()) return
+   
+   
+ !-------------------------------------------------------------------------------------------------
+   !> Read the _Parameters for LIDAR section
+   !------------------------------------------------------------------------------------------------- 
+   
+  CurLine = CurLine + 1  ! Skip section break
+  
+      ! LIDAR Sensor Type
+   CALL ParseVar( InFileInfo, CurLine, "SensorType", InputFileData%SensorType, TmpErrStat, TmpErrMsg, UnEc )
+   CALL SetErrStat( TmpErrStat, TmpErrMsg, ErrStat, ErrMsg, RoutineName )
+   IF (Failed()) RETURN
+
+     ! Number of Range Gates
+   CALL ParseVar( InFileInfo, CurLine, "NumPulseGate", InputFileData%NumPulseGate, TmpErrStat, TmpErrMsg, UnEc )
+   CALL SetErrStat( TmpErrStat, TmpErrMsg, ErrStat, ErrMsg, RoutineName )
+   IF (Failed()) RETURN
+     
+     ! Before proceeding, make sure that NumPulseGate makes sense
+   IF ( InputFileData%NumPulseGate < 0 .OR. InputFileData%NumPulseGate > 5 ) THEN
+      CALL SetErrStat( ErrID_Fatal, 'NumPulseGate must be greater than or equal to zero and less than 5.', &
+                        ErrStat, ErrMsg, RoutineName )
+      RETURN
+   ENDIF
+
+     ! Pulse Gate Spacing
+   CALL ParseVar( InFileInfo, CurLine, "PulseSpacing", InputFileData%PulseSpacing, TmpErrStat, TmpErrMsg, UnEc )
+   CALL SetErrStat( TmpErrStat, TmpErrMsg, ErrStat, ErrMsg, RoutineName )
+   IF (Failed()) RETURN
+
+     ! NumBeam: Number of points to output the lidar measured wind velocity (1 to 5)
+   CALL ParseVar( InFileInfo, CurLine, "NumBeam", InputFileData%NumBeam, TmpErrStat, TmpErrMsg, UnEc )
+   CALL SetErrStat( TmpErrStat, TmpErrMsg, ErrStat, ErrMsg, RoutineName )
+   if (Failed()) return
+
+      ! Before proceeding, make sure that NumBeam makes sense
+    IF ( InputFileData%NumBeam < 1 .OR. InputFileData%NumBeam > 5 ) THEN
+      CALL SetErrStat( ErrID_Fatal, 'NumBeam must be greater than or equal to one and less than 6.', &
+                        ErrStat, ErrMsg, RoutineName )
+      RETURN
+   ENDIF
+   
+       ! Before proceeding, make sure that NumBeam makes sense
+   IF ( InputFileData%NumBeam > 1 .AND. InputFileData%SensorType > 1) THEN
```

Comment:
I would perform this check when transferring data from the input file into the InflowWind data structure. This function should just concentrate on reading the file, so it should validate NumBeam for allocating an array which needs to be done for the following lines.